### PR TITLE
Fix deployment state cleanup

### PR DESCRIPTION
**Pull request by sachscode**
The deployment-state.json is created during "iterative updates" in GraphQL schema deployments. This file is not deleted after completion of the deployment.  Future iterative updates fail if the deployment-state.json file exists but is not in "IDLE" state. 

#### Description of changes
The fix deletes the deployment-state.json file after the completion of any deployment ( success or failure ) . 

#### Issue #, if available
#9665 

#### Description of how you validated changes
manual tests:
1. Force iterative updates on a GraphQL schema.
2. Check deployment-state.json is getting created when deployment is in progress.
3. Validate that deployment-state.json is deleted after deployment concludes.

integration test:
schema-iterative-update-locking test succeeds

#### Checklist

- [*] PR description included
- [*] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-provider-awscloudformation/src/aws-utils/aws-s3.ts [hunk 1]
```diff
@@ -46,6 +74,9 @@ export class S3 {
     this.s3 = new aws.S3({ ...cred, ...options });
   }
 
+  /**
+   *
```
Comment (johnpc) [inline packages/amplify-provider-awscloudformation/src/aws-utils/aws-s3.ts:78]: do we need jsdocs on private functions

File: packages/amplify-provider-awscloudformation/src/aws-utils/aws-s3.ts [hunk 2]
```diff
@@ -71,13 +105,16 @@ export class S3 {
     return s3Params;
   }
 
-  async uploadFile(s3Params: $TSAny, showSpinner: boolean = true) {
+  /**
+   *
```
Comment (johnpc) [inline packages/amplify-provider-awscloudformation/src/aws-utils/aws-s3.ts:109]: we should probably have jsdocs on public functions though

**Comment by johnpc** [comment]
the actual cleanup code lgtm, just a matter of nits wrestling with the linter. approved

**Review by johnpc — APPROVED** [review]

File: packages/amplify-provider-awscloudformation/src/push-resources.ts [hunk 3]
```diff
@@ -303,13 +322,14 @@ export async function run(context: $TSContext, resourceDefinition: $TSObject, re
...
-          } else {
-            throw err;
           }
+          throw err;
         } finally {
           spinner.stop();
         }
       }
+      // Cleanup the deployment-state file
+      await deploymentStateManager.deleteDeploymentStateFile();
```
[the remark this answers is not in the document]
Comment (sachscode) [reply packages/amplify-provider-awscloudformation/src/push-resources.ts:334]: Added a guard inside this function.

**Review by SwaySway — APPROVED** [review]

**Pull request merged**